### PR TITLE
Stop more checks if invalid resources are found

### PR DESCRIPTION
**Pull request by KevinHock**
Analyzing a policy with an INVALID_ARN gives exceptions from all of the community auditors 

e.g.
```json
{
  "Version":"2012-10-17",
  "Statement":[
    {
      "Sid":"AddPerm",
      "Effect":"Allow",
      "Principal": "*",
      "Action":["ssm:PutParameter"],
      "Resource":[{"Fn::Sub": "arn:aws:ssm:*:${AWS::AccountId}:*"}]
    }
  ]
}
```
from [test_resources.py](https://github.com/duo-labs/parliament/blob/c7ea5b25c4b279036fbe773ecb179c9667ba66ea/tests/unit/test_resources.py#L12-L23)

Also:
* Update managed policies repo link to @z0ph's

* Fix some spelling

### CODE REVIEW

**Comment by KevinHock** [comment]
Great project by the way!

Glad I could try to contribute like @danielpops and @piax93, small world.

File: README.md [hunk 1]
```diff
@@ -41,7 +41,7 @@ This example is showing that the action s3:GetObject requires a resource matchin
 The different input types allowed include:
 - --file: Filename
 - --directory: A directory path, for exmaple: `--directory . --include_policy_extension json --exclude_pattern ".*venv.*"`
-- --aws-managed-policies: For use specifically with the repo https://github.com/SummitRoute/aws_managed_policies
+- --aws-managed-policies: For use specifically with the repo https://github.com/z0ph/aws_managed_policies
```
Comment (0xdabbad00) [inline README.md:44]: Good call since I don't update my repo anymore.

File: parliament/__init__.py [hunk 2]
```diff
@@ -28,12 +28,12 @@ def override_config(override_config_path):
...
 
-    # Over-write the settings
+    # Overwrite the settings
     for finding_type, settings in override_config.items():
         if finding_type not in config:
             config[finding_type] = {}
-        for setting, settting_value in settings.items():
-            config[finding_type][setting] = settting_value
+        for setting, setting_value in settings.items():
+            config[finding_type][setting] = setting_value
```
Comment (0xdabbad00) [inline parliament/__init__.py:36]: Nice catch on the spelling issue

File: parliament/statement.py [hunk 3]
```diff
@@ -641,7 +642,7 @@ def analyze_statement(self):
         """
         Given a statement, look for problems and extract out the parts.
 
-        If it is maformed, return False
+        If it is malformed, return False
```
Comment (0xdabbad00) [inline parliament/statement.py:645]: Nice spelling catch again

File: parliament/statement.py [hunk 4]
```diff
@@ -955,4 +955,4 @@ def analyze_statement(self):
                 "RESOURCE_STAR", detail=sorted(self.resource_star), location=self.stmt
             )
 
-        return True
+        return not has_malformed_resource
```
Comment (0xdabbad00) [inline parliament/statement.py:958]: I think this line is the only change the PR mentioned?

**Comment by 0xdabbad00** [comment]
Thank you @KevinHock! Looks like your PR includes a lot of additional changes. I like the spelling changes and the link change, but I'd rather keep my for loop logic as is, as opposed to using `any` and `all`.  Also I don't know what the difference is for the `defaultdict` changes, but I'd rather keep those as is as well.  I'm guessing some sort of automation made some of those changes?  Can you back out those other changes, or send a PR with only the desired change (I think it is only this line: https://github.com/duo-labs/parliament/pull/159/files#diff-1e5beac911c9f6fdd5a9abd17a5b929690ac686ce3491a571fd0baa109388df8R958 ) and those spelling and link corrections?

**Comment by KevinHock** [comment]
sgtm @0xdabbad00 :) Made the changes

File: parliament/statement.py (hunk 4 above)
Comment (KevinHock) [reply parliament/statement.py:958]: Yup, I added a couple of lines about the 2 other commits now, let me know if you'd prefer I mention it in the PR title.

**Comment by 0xdabbad00** [comment]
LGTM. This can be merged @steiza

**Pull request merged**